spec/queries/search_results_query_spec.rb

```diff
@@ -61,5 +61,65 @@
         expect(search_results_query.search_results.count).to eq 1
       end
     end
+
+    context 'given filter_urls_contains filter' do
+      context 'given filter_match_count at least one filter' do
```

Comment:
These tests do not make sense syntactically (missing verb) and logically (why searching for 1 and 2 occurrences since match_count can be dynamic 🙃 ). Since match_count can be omitted, the following tests would make more sense:

```
context 'given filter_urls_contains filter' do
   context 'given match_count filter is nil' do
      it 'returns only the search results which contain specific  word at least once' do
        #...
     end
   end

   context 'given match_count filter is NOT nil' do
      it 'returns only the search results which contain a specific word at least for the define occurrence match count'  do
        #...
     end
   end

   it 'returns empty when the urls do NOT match a specific word' do
     #...
  end
end
```